modules/test/playwright/tests/commerce/commerce-order-content-web/pendingOrder.spec.ts

```diff
@@ -310,81 +318,292 @@ test('LPD-4174 Sales agent can receive email notifications for new orders placed
+		templateType: 'site-initializer',
+	});
 
-		const products =
-			await apiHelpers.headlessCommerceAdminCatalog.getProductsPage(
-				50,
-				''
-			);
+	apiHelpers.data.push({id: site.id, type: 'site'});
 
-		for (let i = 0; i < products.totalCount; i++) {
-			if (products.items[i].catalogId === catalogs.items[0].id) {
-				apiHelpers.data.push({
-					id: products.items[i].productId,
-					type: 'product',
-				});
-			}
-		}
-
-		const options =
-			await apiHelpers.headlessCommerceAdminCatalog.getOptions();
-
-		for (let i = 0; i < options.totalCount; i++) {
-			apiHelpers.data.push({
-				id: options.items[i].id,
-				type: 'option',
-			});
-		}
-
-		const optionCategories =
-			await apiHelpers.headlessCommerceAdminCatalog.getOptionCategories();
-
-		for (let i = 0; i < optionCategories.totalCount; i++) {
-			apiHelpers.data.push({
-				id: optionCategories.items[i].id,
-				type: 'optionCategory',
-			});
-		}
-
-		const specifications =
-			await apiHelpers.headlessCommerceAdminCatalog.getSpecifications();
-
-		for (let i = 0; i < specifications.totalCount; i++) {
-			apiHelpers.data.push({
-				id: specifications.items[i].id,
-				type: 'specification',
-			});
-		}
-
-		const warehouses =
-			await apiHelpers.headlessCommerceAdminInventoryApiHelper.getWarehousesPage();
-
-		for (let i = 0; i < warehouses.totalCount; i++) {
-			apiHelpers.data.push({
-				id: warehouses.items[i].id,
-				type: 'warehouse',
-			});
-		}
+	const account = await apiHelpers.headlessAdminUser.postAccount({
+		type: 'business',
+	});
+
+	apiHelpers.data.push({id: account.id, type: 'account'});
+
+	const user =
+		await apiHelpers.headlessAdminUser.getUserAccountByEmailAddress(
+			'demo.unprivileged@liferay.com'
+		);
+
+	await apiHelpers.headlessAdminUser.assignUserToAccountByEmailAddress(
+		account.id,
+		['demo.unprivileged@liferay.com']
+	);
+
+	const companyId = await page.evaluate(() => {
+		return Liferay.ThemeDisplay.getCompanyId();
+	});
+
+	const role = await apiHelpers.headlessAdminUser.postRole({
+		name: 'Buyer ' + getRandomString(),
+		rolePermissions: [
+			{
+				actionIds: ['MANAGE_ADDRESSES', 'VIEW_ADDRESSES'],
+				primaryKey: '0',
+				resourceName: 'com.liferay.account.model.AccountEntry',
+				scope: 3,
+			},
+			{
+				actionIds: ['VIEW'],
+				primaryKey: companyId,
+				resourceName: 'com.liferay.commerce.model.CommerceOrderType',
+				scope: 1,
+			},
+			{
+				actionIds: [
+					'ADD_COMMERCE_ORDER',
+					'CHECKOUT_OPEN_COMMERCE_ORDERS',
+					'MANAGE_COMMERCE_ORDER_DELIVERY_TERMS',
+					'MANAGE_COMMERCE_ORDER_PAYMENT_METHODS',
+					'MANAGE_COMMERCE_ORDER_PAYMENT_TERMS',
+					'MANAGE_COMMERCE_ORDER_SHIPPING_OPTIONS',
+					'VIEW_BILLING_ADDRESS',
+					'VIEW_COMMERCE_ORDERS',
+					'VIEW_OPEN_COMMERCE_ORDERS',
+				],
+				primaryKey: '0',
+				resourceName: 'com.liferay.commerce.order',
+				scope: 3,
+			},
+		],
+	});
+
+	await apiHelpers.headlessAdminUser.postRoleUserAccountAssociation(
```

Comment:
you can use the postRoleByExternalReferenceCode where the ERC is the role name (buyer), this way you don't need to create a new role

Reply:
I tried it, but it was not working